app/scripts/controllers/provider-approval.js

```diff
@@ -41,7 +41,7 @@ class ProviderApprovalController extends SafeEventEmitter {
         return
       }
       // register the provider request
-      const metadata = await getSiteMetadata(origin)
+      const metadata = /localhost|[.:/]/.test(origin) ? await getSiteMetadata(origin) : { name: 'An extenstion' }
```

Comment:
I have some reservations about this choice in default extension name. Maybe something like `Unknown browser extension` would be more descriptive, and would convey that we don't know what it is? Open to other ideas too.

Reply:
I like that actually. I was actually worried you and others wouldn't. I'd do "An unknown browser extension".

Reply:
`Unknown` might sound a bit scary to users, but I don't mind encouraging users to be cautious. Plus, it'll act as an incentive for extension authors to add proper metadata (once that's supported at least).

I chose to omit the `An` because of the context in which it's shown - it's styled as a title, not a description.

Though on that point - maybe we're better off choosing a fallback title in the UI rather than here. This feels like the wrong place to make UI decisions. Though the front-end does need to know that this is from an extension, not a normal page.

I've just had an idea - we could bypass the problem of detecting whether it's an extension or not completely *and* avoid making UI decisions in the background by adding `origin` to the metadata. Then on the front-end, you could fallback to showing the origin directly in place of a title if the title is missing.

I think this would be rather straightforward to implement. You'd need to add it to this metadata object, and pass it through to the state update in `approveProviderRequestByOrigin`, and that's about it for the background. In the UI you'd need to update `ui/app/components/app/provider-page-container/provider-page-container-content/` to make `siteTitle` optional, add `origin`, and set `origin` as the fallback. Then there are a few more components above that you'd need to pass the origin through - just search for `siteTitle` to find them.

Reply:
> I chose to omit the An because of the context in which it's shown

Sure, I can drop the 'an'.

> This feels like the wrong place to make UI decisions.

I completely agree. I can modify `_handleProviderRequest` and other downstream modules/functions, to take an additional, but optional, argument `isPossiblyExtension`, which will be handled by the UI to display the message accordingly. This will result in more lines of code for this PR, but it will result in thew UI decision being made in a more appropriate place.

> detecting whether it's an extension

I just want to make it clear that isn't not possible to detect that something _is_ an extension (without the management api) and thus it is only possible to detect if something _definitely isn't_ an extension.

> by adding origin to the metadata

Yes, either the UI can detect with regex if the origin isn't an extension by regex, or as I mentioned in my previous comment, with a `isPossiblyExtension` bool passed to it as an argument or in the metadata object. However, this does not solve the issue that if you try to call `getSiteMetadata` when origin is an extension, the function will crash and you will not be able to proceed, so you still need to regex before calling `getSiteMetadata`.

So, at the end of the day, while I can do what you suggested, I cannot call `getSiteMetadata` when `origin` is an extension ID, or else it will crash (as is current state).